packages/components/src/composite/test/index.tsx

```diff
@@ -51,25 +28,142 @@ async function renderAndValidate( ...args: Parameters< typeof render > ) {
 	return view;
 }
 
-function RemoveItemTest( props: ComponentProps< typeof Composite > ) {
-	const [ showThirdItem, setShowThirdItem ] = useState( true );
-	return (
-		<>
-			<button>Focus trap before composite</button>
-			<Composite { ...props }>
+describe( 'Composite', () => {
+	let clientHeightSpy: jest.SpiedGetter<
+		typeof HTMLElement.prototype.clientHeight
+	>;
+
+	beforeAll( () => {
+		// This is necessary because of how Ariakit calculates page up and
+		// page down. Without this, nothing has a height, and so paging up
+		// and down doesn't behave as expected in tests.
+		clientHeightSpy = jest
+			.spyOn( HTMLElement.prototype, 'clientHeight', 'get' )
+			.mockImplementation( function getClientHeight( this: HTMLElement ) {
+				if ( this.tagName === 'BODY' ) {
+					return window.outerHeight;
+				}
+				return 50;
+			} );
+	} );
+
+	afterAll( () => {
+		clientHeightSpy?.mockRestore();
+	} );
+
+	test( 'Renders as a single tab stop', async () => {
+		await renderAndValidate(
+			<>
+				<button>Before</button>
+				<Composite>
+					<Composite.Item>Item 1</Composite.Item>
+					<Composite.Item>Item 2</Composite.Item>
+					<Composite.Item>Item 3</Composite.Item>
+				</Composite>
+				<button>After</button>
+			</>
+		);
+
+		await press.Tab();
+		expect( screen.getByText( 'Before' ) ).toHaveFocus();
+		await press.Tab();
+		expect( screen.getByText( 'Item 1' ) ).toHaveFocus();
+		await press.Tab();
+		expect( screen.getByText( 'After' ) ).toHaveFocus();
+		await press.ShiftTab();
+		expect( screen.getByText( 'Item 1' ) ).toHaveFocus();
```

Comment:
Did you want these to use `getByRole` as well?

Reply:
Well spotted, doing it in https://github.com/WordPress/gutenberg/pull/65952/commits/72bc7d63ed652d404bcabffd48ea7255ea3398d4